bin/tests/check_dl_callbacks.sh

```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/env bash
+
+# Cause the script to exit if a single command fails
+set -eo pipefail -v
+
+
+################################  global variables  ################################
+rm -rf ./tests/logs
+
+EXPDIR=./tests/_tests_dl_callbacks
+LOGDIR=./tests/logs/_tests_dl_callbacks
+CHECKPOINTS=${LOGDIR}/checkpoints
+LOGFILE=${CHECKPOINTS}/_metrics.json
+
+
+function check_file_existence {
+    if [[ ! -f $1 ]]
+    then
+        echo "There is no '$1'!"
+        exit 1
+    fi
+}
+
+
+function check_num_files {
+    NFILES=$( ls $1 | wc -l )
+    if [[ $NFILES -ne $2 ]]
+    then
+        echo "Different number of files in '$1' - expected $2 but actual number is $NFILES!"
+        exit 1
+    fi
+}
+
+
+function check_checkpoints {
+    check_num_files "${CHECKPOINTS}/${1}.pth" $2
+    check_num_files "${CHECKPOINTS}/${1}_full.pth" $2
+}
+
+
+################################  pipeline 00  ################################
+# checking dafult parameters of checkpoint and one stage
+LOG_MSG='pipeline 00'
+echo ${LOG_MSG}
+
+PYTHONPATH=./examples:./catalyst:${PYTHONPATH} \
+  python catalyst/dl/scripts/run.py \
+  --expdir=${EXPDIR} \
+  --config=${EXPDIR}/config0.yml \
+  --logdir=${LOGDIR}
+
+if [[ ! (-f "$LOGFILE" && -r "$LOGFILE") ]]; then
+    echo "File $LOGFILE does not exist"
+    exit 1
+fi
+
+cat ${LOGFILE}
+echo ${LOG_MSG}
+
+check_file_existence ${LOGFILE}
+check_checkpoints "best" 1
+check_checkpoints "last" 1
+check_checkpoints "stage1\.[[:digit:]]" 1
+check_num_files ${CHECKPOINTS} 7
```

Comment:
please add a comment, that `we have 3x2 checkpoints + one metric.json file` to make it easier to understand "magic numbers"